src/commands.cpp

```diff
@@ -7007,5 +7008,186 @@ void CommandSMSVerificationCheck::procresult()
     }
 };
 
+void CommandGetRegisteredContacts::processResult(MegaApp& app, JSON& json)
+{
+    if (json.isnumeric())
+    {
+        app.getregisteredcontacts_result(static_cast<error>(json.getint()), nullptr);
+        return;
+    }
+
+    vector<tuple<string, string, string>> registered_contacts;
+
+    string entry_user_detail;
+    string id;
+    string user_detail;
+
+    for (;;)
+    {
+        if (json.enterobject())
+        {
+            bool end_of_object = false;
+            while (!end_of_object)
+            {
+                switch (json.getnameid())
+                {
+                    case MAKENAMEID3('e', 'u', 'd'):
+                    {
+                        json.storeobject(&entry_user_detail);
+                        break;
+                    }
+                    case MAKENAMEID2('i', 'd'):
+                    {
+                        json.storeobject(&id);
+                        break;
+                    }
+                    case MAKENAMEID2('u', 'd'):
+                    {
+                        json.storeobject(&user_detail);
+                        break;
+                    }
+                    case EOO:
+                    {
+                        end_of_object = true;
+                        break;
+                    }
+                    default:
+                    {
+                        if (!json.storeobject())
+                        {
+                            LOG_err << "Failed to parse 'get registered contacts' response";
+                            app.getregisteredcontacts_result(API_EINTERNAL, nullptr);
+                            return;
+                        }
+                    }
+                }
+            }
+            json.leaveobject();
+            if (entry_user_detail.empty() || id.empty() || user_detail.empty())
+            {
+                LOG_err << "Missing or empty field when parsing 'get registered contacts' response";
+                app.getregisteredcontacts_result(API_EINTERNAL, nullptr);
+                return;
+            }
+            registered_contacts.emplace_back(move(entry_user_detail), move(id), move(user_detail));
+        }
+        else
+        {
+            break;
+        }
+    }
+    app.getregisteredcontacts_result(API_OK, &registered_contacts);
+}
+
+void CommandGetRegisteredContacts::procresult()
+{
+    processResult(*client->app, client->json);
+}
+
+CommandGetRegisteredContacts::CommandGetRegisteredContacts(MegaClient* client, const MegaStringMap& contacts)
```

Comment:
The pattern in this .cpp file is that the constructor is first, then the procresult() after - I've been pulled up on that one :-)

Reply:
That is weird. Shouldn't the order in the cpp match the header??

Reply:
Well, it's the pattern everyone is used to, I guess

Reply:
Will fix